plugins/inputs/snmp/snmp.go

```diff
@@ -857,53 +878,31 @@ func snmpTableCall(oid string) (mibName string, oidNum string, oidText string, f
 	}
 
 	mibPrefix := mibName + "::"
-	oidFullName := mibPrefix + oidText
 
 	// first attempt to get the table's tags
 	tagOids := map[string]struct{}{}
 	// We have to guess that the "entry" oid is `oid+".1"`. snmptable and snmptranslate don't seem to have a way to provide the info.
-	if out, err := execCmd("snmptranslate", "-Td", oidFullName+".1"); err == nil {
-		scanner := bufio.NewScanner(bytes.NewBuffer(out))
-		for scanner.Scan() {
-			line := scanner.Text()
-
-			if !strings.HasPrefix(line, "  INDEX") {
-				continue
-			}
-
-			i := strings.Index(line, "{ ")
-			if i == -1 { // parse error
-				continue
-			}
-			line = line[i+2:]
-			i = strings.Index(line, " }")
-			if i == -1 { // parse error
-				continue
-			}
-			line = line[:i]
-			for _, col := range strings.Split(line, ", ") {
-				tagOids[mibPrefix+col] = struct{}{}
-			}
-		}
-	}
+	// mimcks grabbing INDEX {} that is returned from snmptranslate -Td MibName
+	submask := oidNum + ".1"
+	node, err := gosmi.GetNodeByOID(types.OidMustFromString(submask))
 
-	// this won't actually try to run a query. The `-Ch` will just cause it to dump headers.
-	out, err := execCmd("snmptable", "-Ch", "-Cl", "-c", "public", "127.0.0.1", oidFullName)
 	if err != nil {
-		return "", "", "", nil, fmt.Errorf("getting table columns: %w", err)
+		return "", "", "", nil, fmt.Errorf("getting submask: %w", err)
 	}
-	scanner := bufio.NewScanner(bytes.NewBuffer(out))
-	scanner.Scan()
-	cols := scanner.Text()
-	if len(cols) == 0 {
-		return "", "", "", nil, fmt.Errorf("could not find any columns in table")
+
+	index := node.GetIndex()
+
+	for i := range index {
+		tagOids[mibPrefix+index[i].Name] = struct{}{}
 	}
-	for _, col := range strings.Split(cols, " ") {
-		if len(col) == 0 {
-			continue
-		}
-		_, isTag := tagOids[mibPrefix+col]
-		fields = append(fields, Field{Name: col, Oid: mibPrefix + col, IsTag: isTag})
+
+	// grabs all columns from the table
+	// mimmicks grabbing everything returned from snmptable -Ch -Cl -c public 127.0.0.1 oidFullName
+	col := node.GetRow().AsTable().ColumnOrder
+
+	for i := range col {
+		_, isTag := tagOids[mibPrefix+col[i]]
+		fields = append(fields, Field{Name: col[i], Oid: mibPrefix + col[i], IsTag: isTag})
```

Comment:
Any reason to not use the index/value pair directly?
```suggestion
	for i,c  := range col {
		_, isTag := tagOids[mibPrefix+c]
		fields = append(fields, Field{Name: c, Oid: mibPrefix + c, IsTag: isTag})
```

Reply:
I didn't think of that, good catch